prov/verbs/src/verbs_info.c

```diff
@@ -1911,21 +1894,49 @@ static void vrb_filter_info_by_addr_format(struct fi_info **info, int addr_forma
 	}
 }
 
+void vrb_devs_free(struct dlist_entry *verbs_devs)
+{
+	struct verbs_dev_info *dev;
+	struct verbs_addr *addr;
+
+	while (!dlist_empty(verbs_devs)) {
+		dlist_pop_front(verbs_devs, struct verbs_dev_info, dev, entry);
+		while (!dlist_empty(&dev->addrs)) {
+			dlist_pop_front(&dev->addrs, struct verbs_addr, addr, entry);
+			rdma_freeaddrinfo(addr->rai);
+			free(addr);
+		}
+		free(dev->name);
+		free(dev);
+	}
+}
+
 int vrb_getinfo(uint32_t version, const char *node, const char *service,
 		   uint64_t flags, const struct fi_info *hints,
 		   struct fi_info **info)
 {
+	static bool init_done = false;
 	int ret;
 
 	vrb_prof_func_start(__func__);
-	ofi_mutex_lock(&vrb_info_mutex);
-	ret = vrb_init_info(&vrb_util_prov.info);
-	if (ret) {
-		ofi_mutex_unlock(&vrb_info_mutex);
-		goto out;
+	if (!init_done || flags & FI_RESCAN) {
+		if (!init_done) {
+			ret = vrb_init();
+			if (ret) {
+				ofi_mutex_unlock(&vrb_info_mutex);
```

Comment:
The mutex unlock does not have a matching lock.

Reply:
Fixed. Thanks